clippy_lints/src/len_zero.rs

```diff
@@ -143,7 +143,7 @@ impl<'tcx> LateLintPass<'tcx> for LenZero {
             && let Some(ty_id) = cx.qpath_res(ty_path, imp.self_ty.hir_id).opt_def_id()
             && let Some(local_id) = ty_id.as_local()
             && let ty_hir_id = cx.tcx.local_def_id_to_hir_id(local_id)
-            && !is_lint_allowed(cx, LEN_WITHOUT_IS_EMPTY, ty_hir_id)
+            && !fulfill_or_allowed(cx, LEN_WITHOUT_IS_EMPTY, [ty_hir_id, item.hir_id()])
```

Comment:
It also needs to be moved to before the `span_lint` otherwise we could fulfil `#[expect]`s that wouldn't actually lint. e.g. currently I think it would fulfil an `#[expect(clippy::len_without_is_empty)]` on `fn len` even if there was as a `fn is_empty`

Reply:
Of course. Fixed and checked.